Remove libwinpthread-1.dll dependency

d0k3 opened the pull request:
Without this commit, ctrtool and makerom won't run on Windows systems that don't have libwinpthread-1.dll.

jakcron commented on makerom/Makefile:
```diff
@@ -8,7 +8,7 @@ CC = gcc
     #Windows Build CFG
     CFLAGS += -Wno-unused-but-set-variable
-    LIBS += -static-libgcc
+    LIBS += -static -static-libgcc -static-libstdc++ -lpthread -L libwinpthread-1.dll
```
Why is -static-libstdc++ here? There's no c++ code in makerom.

jakcron commented:
Really? The binaries I compile are the same size before and after the commit. Additionally my binaries run on Windows without the libwinpthread.dll, or any windows development environment.

profi200 commented:
What do you use to build this? Compiling with the previous flags makes executables which run perfectly on any Windows machine. Compiler is MinGW-W64.

d0k3 commented:
I'm using MINGW-W64 (GCC 4.9.2). If you want to have a look, this archive contains compiled executables before and after my change:
http://wikisend.com/download/861210/Project_CTR-d0k3.zip

Also see this Dependency Walker Screenshot (for ctrtool):
http://imgur.com/ouSvQb5

The change for makerom, on the other hand, seems to be unneeded. I reverted it and squashed the change back into the original commit.

jakcron commented:
@profi200 My personal builds of ctrtool do not include dependency on libwinpthread-1.dll.

d0k3 is using "GCC: (i686-posix-dwarf-rev0, Built by MinGW-W64 project) 4.9.2" to build these. 

This appears to be a matter of having "just the right" compiler setup. I'll let you make the call on what should be done.

profi200 commented:
We both use this one "gcc (x86_64-win32-seh-rev1, Built by MinGW-W64 project) 5.2.0" which could explain the unnecessary dependency. It uses win32 threading instead of posix.

d0k3 commented:
Well, is this because my GCC is too outdated or because I'm using the POSIX version? I haven't specifically selected to download the POSIX version and have just used the standard MINGW installer to download everything?

The issue itself isn't exactly a rare one:
https://www.google.de/search?q=libwinpthread-1.dll

Also, does that change in the Makefile have any effect on how your executables are built?

profi200 commented:
This is how it looks for ctrtool here with MinGW-W64 and latest source:

https://i.imgur.com/Yvqkehd.png

d0k3 commented:
Okay, I just reinstalled MinGW using the official installer from here:
http://sourceforge.net/projects/mingw-w64/

The default setting is posix, not win32, thus generating that libwinpthread-1.dll dependency for certain source files. Of course, I made sure this time to select win32, not posix. And, yup, with the MinGW installation that dependency is not there.
